src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -255,6 +257,66 @@ private static string ReplaceWith24HourClock(string format)
             return Regex.Replace(format, pattern, "HH${nonHours}", RegexOptions.IgnorePatternWhitespace);
         }
 
+        /// <summary>
+        /// Resolves month/minute ambiguity and 24h vs AM/PM.
+        /// </summary>
+        /// <param name="format">Format string.</param>
+        /// <returns></returns>
+        private static string ResolveDateTimeFormatAmbiguities(string format)
```

Comment:
>ResolveDateTimeFormatAmbiguities [](http://example.com/codeflow?start=30&length=32)

This looks like a complicated function ... can we create malformed inputs that will cause it to crash? or take long times?

Reply:
JS test cases have been provided with some samples of formats.